ksp_plugin/manœuvre_body.hpp

```diff
@@ -0,0 +1,131 @@
+﻿#pragma once
+
+#include "ksp_plugin/manœuvre.hpp"
+
+#include <cmath>
+
+#include "quantities/elementary_functions.hpp"
+
+namespace principia {
+
+using quantities::Sqrt;
+
+namespace ksp_plugin {
+
+template <typename Frame>
+Manœuvre<Frame>::Manœuvre(Force thrust,
+                          Mass initial_mass,
+                          Speed effective_exhaust_velocity,
+                          Vector<double, Frame> direction)
+    : thrust_(thrust),
+      initial_mass_(initial_mass),
+      effective_exhaust_velocity_(effective_exhaust_velocity),
+      direction_(direction) {}
+
+template<typename Frame>
+Instant Manœuvre<Frame>::initial_time() const {
+  CHECK(initial_time_ != nullptr);
+  return *initial_time_;
+}
+
+template<typename Frame>
+inline Instant Manœuvre<Frame>::time_of_half_Δv() const {
+  return initial_time() + time_to_half_Δv();
+}
+
+template<typename Frame>
+Instant Manœuvre<Frame>::final_time() const {
+  return initial_time() + duration();
+}
+
+template<typename Frame>
+void Manœuvre<Frame>::set_initial_time(Instant const & initial_time) {
+  initial_time_ = std::make_unique<Instant>(initial_time);
+}
+
+template <typename Frame>
+void Manœuvre<Frame>::set_time_of_half_Δv(Instant const& time_of_half_Δv) {
+  set_initial_time(time_of_half_Δv - time_to_half_Δv());
+}
+
+template<typename Frame>
+inline Time Manœuvre<Frame>::duration() const {
+  CHECK(duration_ != nullptr);
+  return *duration_;
+}
+
+template<typename Frame>
+void Manœuvre<Frame>::set_duration(Time const & duration) {
+  duration_ = std::make_unique<Time>(duration);
+}
+
+template<typename Frame>
+void Manœuvre<Frame>::set_Δv(Speed const& Δv) {
+  set_duration(initial_mass() * effective_exhaust_velocity() *
+               (1 - std::exp(-Δv / effective_exhaust_velocity())) / thrust());
```

Comment:
We want this to be independent of implementation (so we can change which members are stored).